### PR TITLE
Update query-execution-plan.md (#3359)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #3359 to release-4.0

---

### What is changed, added or deleted? (Required)

Update the documentation for query-execution-plan.md

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3201/ , part of [ pingcap/docs-cn#3170](https://github.com/pingcap/docs-cn/pull/3170)
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [x] Delete files
- [x] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: query-execution-plan.md [hunk 1]
```diff
@@ -1,16 +1,17 @@
 ---
 title: Understand the Query Execution Plan
 summary: Learn about the execution plan information returned by the `EXPLAIN` statement in TiDB.
-aliases: ['/docs/stable/query-execution-plan/','/docs/v4.0/query-execution-plan/','/docs/stable/reference/performance/understanding-the-query-execution-plan/']
+aliases: ['/docs/dev/query-execution-plan/','/docs/dev/reference/performance/understanding-the-query-execution-plan/','/docs/dev/index-merge/','/docs/dev/reference/performance/index-merge/','/tidb/dev/index-merge']
+>>>>>>> 2e2916f... Update query-execution-plan.md (#3359)
```
Comment (ran-huang) [inline query-execution-plan.md:5]: @ireneontheway Please delete this line as well~

Comment (ran-huang) [reply query-execution-plan.md:5]: Also, the bot failed to delete `index-merge.md`. Please delete this, too.

Comment (ireneontheway) [reply query-execution-plan.md]: Done~

File: query-execution-plan.md [hunk 2]
```diff
@@ -1,16 +1,16 @@
 ---
 title: Understand the Query Execution Plan
 summary: Learn about the execution plan information returned by the `EXPLAIN` statement in TiDB.
-aliases: ['/docs/stable/query-execution-plan/','/docs/v4.0/query-execution-plan/','/docs/stable/reference/performance/understanding-the-query-execution-plan/']
+aliases: ['/docs/stable/query-execution-plan/','/docs/stable/reference/performance/understanding-the-query-execution-plan/','/docs/stable/index-merge/','/docs/stable/reference/performance/index-merge/','/tidb/stable/index-merge']
```
Comment (yikeke) [inline query-execution-plan.md:4]: There should 7 aliases: @ran-huang @ireneontheway 

['/docs/stable/query-execution-plan/','/docs/v4.0/query-execution-plan/','/docs/stable/reference/performance/understanding-the-query-execution-plan/','/docs/stable/index-merge/','/docs/v4.0/index-merge/','/docs/stable/reference/performance/index-merge/','/tidb/stable/index-merge']

Comment (ran-huang) [reply query-execution-plan.md:4]: fixed in https://github.com/pingcap/docs/pull/3429

**Pull request merged**